core/src/test/scala/kafka/tools/GetOffsetShellTest.scala

```diff
@@ -166,6 +166,13 @@ class GetOffsetShellTest extends KafkaServerTestHarness with Logging {
     )
   }
 
+  @Test
+  def testNoOffsetIfTimestampGreaterThanLatestRecord(): Unit = {
```

Comment:
Checking the code, it looks like the unknown_offset could mean the error state or, like your test did, the timestamp is not found. Could we also add a test to verify if the request has some error response, we should get empty offset, too (due to the unknown_offset response). Ex: the topic doesn't exist.

Reply:
OK, I see